src/flow-manager.c

```diff
@@ -248,6 +249,26 @@ static int FlowManagerFlowTimeout(const Flow *f, enum FlowState state, struct ti
 
     /* do the timeout check */
     if (flow_times_out_at >= ts->tv_sec) {
+        /* for bypassed flow we need to check if the capture method has seen
+         * something and update the logic */
+        FlowBypassInfo *fc = FlowGetStorageById(f, GetFlowBypassInfoID());
+        if (fc) {
```

Comment:
```if (fc && fc->BypassUpdate) {```

Reply:
can this whole logic move to a (inline) util func? Fine if it stays in this file.